### ISSUE TITLE
Allow for custom port

### ISSUE THREAD

**Issue by katossky** [eth-sri/lmql#34]
When serving the model on a custom port with
```
lmql serve-model bigscience/bloom-7b1 --port 8031
```
... there doesn't seem to be any way to redirect the Python client towards the correct port with the decorator `@lmql.query`.

The `8080` port seems hard coded in several places of the code: https://github.com/eth-sri/lmql/search?q=8080

**Comment by katossky** [comment]
I am willing to help, with some guidance

After inspection, it looks like the [`load`](https://github.com/eth-sri/lmql/blob/47ca14434cc1bb3361e9174979f3f613e24a6f16/src/lmql/__init__.py#L79) instruction inside `run_file` has a `autoconnect=True` option (that we can't modify from higher-level functions), which launches [`_autoconnect_model`](https://github.com/eth-sri/lmql/blob/47ca14434cc1bb3361e9174979f3f613e24a6f16/src/lmql/__init__.py#L63-L64) which has [localhost-8080 hard coded](https://github.com/eth-sri/lmql/blob/47ca14434cc1bb3361e9174979f3f613e24a6f16/src/lmql/__init__.py#L50).

**Comment by lbeurerkellner** [comment]
Yes, we definitely should change this, thanks for the suggestion and offering help. 

The solution we thought about is to allow strings like `<model>@<host>:<port>` so e.g. `stabilityai/stablelm-tuned-alpha-7b@localhost:1234", parse host/port in the function you mention and pass everything before `@` to the existing API logic. 

What do you think? The files you were looking at are the correct place.

**Comment by katossky** [comment]
You prefer that to adding a `port` and `host` parameter to `lmql.query` ?

**Comment by lbeurerkellner** [comment]
The updated inference infrastructure now allows custom ports and host with `lmql serve-model`.
